PosteAPI/views.py

```diff
@@ -83,6 +81,29 @@ def post(self, request, *args, **kwargs):
                 {"result": {"success": False, "errors": serializer.errors}},
                 status=status.HTTP_400_BAD_REQUEST,
             )
+          
+    def update(self, request, *args, **kwargs):
+        user = request.user
+        data = request.data
+        folder = Folder.objects.get(['folder_id'])
+        if not user.can_share_folder(folder):
+            return Response({"detail": "You do not have permission to share this folder."},
+                            status=status.HTTP_403_FORBIDDEN)
+
+        permission, created = FolderPermission.objects.get_or_create(
+            user=User.objects.get(data['user_id']),
```

Comment:
Might be helpful to have the application pass in the token acquired from login for auth here, as opposed to a user id